pdp/handlers.go

```diff
@@ -974,7 +974,30 @@ func (p *PDPService) handleAddPieceToDataSet(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	// Step 9: Insert into message_waits_eth and pdp_data_set_pieces
+	// Step 9: check for indexing requirements on data set.
+	// Get listenerAddr from blockchain contract
+	pdpVerifier, err := contract.NewPDPVerifier(contract.ContractAddresses().PDPVerifier, p.ethClient)
+	if err != nil {
+		log.Errorw("Failed to instantiate PDPVerifier contract", "error", err, "dataSetId", dataSetId)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+	listenerAddr, err := pdpVerifier.GetDataSetListener(nil, dataSetId)
+	if err != nil {
+		log.Errorw("Failed to get listener address for data set", "error", err, "dataSetId", dataSetId)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+	mustIndex, _, err := contract.GetDataSetMetadataAtKey(listenerAddr, p.ethClient, dataSetId, "withIPFSIndexing")
+	if err != nil {
+		// Hard to differenctiate between unsupported listener type OR internal error
+		// So we log on debug and skip indexing attempt
```

Comment:
this is what I think you should move up into `GetDataSetMetadataAtKey` - we can differentiate so let's just do it there